### PR TITLE
feat: personas in new dashboard

### PR DESCRIPTION
**Pull request by nuanyang233**
figma: https://www.figma.com/file/xxHFHHzRgN2E90xCOB83ae/Dashboard?node-id=2207%3A79396&viewport=-6071%2C2311%2C0.7068276405334473

### CODE REVIEW

File: packages/dashboard/src/pages/routes.tsx [hunk 1]
```diff
@@ -23,7 +24,11 @@ export function Pages() {
         <Suspense fallback="loading...">
             <Switch>
                 <Route path={Routes.Welcome} children={<Welcome />} />
-                <Route path={Routes.Personas} children={frame(<Personas />)} exact />
+                <Route
+                    path={Routes.Personas}
+                    children={<PersonaState.Provider>{frame(<Personas />)}</PersonaState.Provider>}
```
Comment (Jack-Works) [inline packages/dashboard/src/pages/routes.tsx:29]: I think it's better to add inside `<Personas />`

File: packages/dashboard/src/pages/Personas/components/DeletePersonaDialog/index.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import { alpha, Button, InputBase, Link, makeStyles, Typography } from '@material-ui/core'
+import { createStyles, experimentalStyled as styled } from '@material-ui/core/styles'
+import type { Persona } from '../../../../../../maskbook/src/database'
+import { memo } from 'react'
+import { MaskColorVar, MaskDialog } from '../../../../../../theme'
```
Comment (Jack-Works) [inline packages/dashboard/src/pages/Personas/components/DeletePersonaDialog/index.tsx:5]:
```suggestion
import { MaskColorVar, MaskDialog } from '@dimensiondev/maskbook-theme'
```

File: packages/dashboard/src/pages/Personas/hooks/useConnectSocialNetwork.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+export interface SocialNetworkProvider {
+    internalName: string
+    network: string
+    connected: boolean
+}
+
+export function useConnectSocialNetwork() {
+    return useAsyncFn(async (identifier: string, provider: SocialNetworkProvider) => {
+        return Services.SocialNetwork.connectSocialNetwork(identifier, provider)
+    })
```
Comment (Jack-Works) [inline packages/dashboard/src/pages/Personas/hooks/useConnectSocialNetwork.ts:13]: It seems `connectSocialNetwork` returns `Promise<void>`. Why not just call it directly?

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: packages/dashboard/package.json [hunk 4]
```diff
@@ -20,10 +20,15 @@
     "@dimensiondev/plugin-example": "workspace:*",
     "@types/color": "^3.0.1",
     "async-call-rpc": "^5.0.0",
+    "classNames": "npm:classnames@^2.2.6",
```
[the remark this answers is not in the document]
Comment (septs) [reply packages/dashboard/package.json:23]: plz revert the rename.

File: packages/theme/src/constants.ts [hunk 5]
```diff
@@ -34,20 +35,24 @@ export const LightColor = {
...
-    primaryContrastText: 'white',
+    primaryContrastText: '#ffffff',
 
     secondary: '#242e57',
     secondaryContrastText: '#1c68f3',
 
     primaryBackground: '#212442',
     secondaryBackground: '#252846',
-
+    suspensionBackground: 'rgba(27, 30, 60, 0.8)',
```
[the remark this answers is not in the document]
Comment (nuanyang233) [reply packages/theme/src/constants.ts:51]: The theme color in figma

File: packages/maskbook/src/extension/background-script/SocialNetworkService.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import { requestSNSAdaptorPermission } from '../../social-network/utils/permissions'
+
+import { delay } from '../../utils/utils'
+import { currentSetupGuideStatus } from '../../settings/settings'
+import stringify from 'json-stable-stringify'
+import { SetupGuideStep } from '../../components/InjectedComponents/SetupGuide'
+
+export const getDefinedSocialNetworkUIs = async () => {
+    return definedSocialNetworkUIs
+}
```
[the remark this answers is not in the document]
Comment (nuanyang233) [reply packages/maskbook/src/extension/background-script/SocialNetworkService.ts:12]: Ok, But I need it in dashborad...So What should I do

[the remark this answers is not in the document]
Comment (Jack-Works) [reply packages/maskbook/src/extension/background-script/SocialNetworkService.ts:12]: You should only provide the information you need. `definedSocialNetworkUIs` contains functions in its data structure.

[the remark this answers is not in the document]
Comment (nuanyang233) [reply packages/maskbook/src/extension/background-script/SocialNetworkService.ts:12]: Like `return defiendSocialNetworkUIs.get(key)`?

File: packages/dashboard/src/pages/routes.tsx (hunk 1 above)
Comment (nuanyang233) [reply packages/dashboard/src/pages/routes.tsx:29]: I'm afraid this won't work, as `PersonaState.useContainer()` is used in `Personas`. So the Provider wraps the `Personas` component

File: packages/dashboard/src/pages/Personas/hooks/useConnectSocialNetwork.ts (hunk 3 above)
Comment (nuanyang233) [reply packages/dashboard/src/pages/Personas/hooks/useConnectSocialNetwork.ts:13]: The `loading` state may be used later to display the loading indicator

File: packages/maskbook/src/extension/background-script/SocialNetworkService.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (Jack-Works) [reply packages/maskbook/src/extension/background-script/SocialNetworkService.ts:12]: No. You should rephrase what you need and pick only those values from the SNSAdaptor.

File: packages/dashboard/src/pages/Personas/components/AddPersonaCard/index.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+            gridColumnGap: theme.spacing(3),
+        },
+    }),
+)
+
+const NameInput = styled(InputBase)(({ theme }) => ({
+    'label + &': {
+        marginTop: theme.spacing(3),
+    },
+    '& .MuiInputBase-input': {
```
Comment (Jack-Works) [inline packages/dashboard/src/pages/Personas/components/AddPersonaCard/index.tsx:26]: Please don't use `.MuiInputBase-input`

File: packages/shared/src/index.ts [hunk 8]
```diff
@@ -10,3 +10,5 @@ export * from './ShadowRoot'
 export * from './TypedMessage'
 export * from './utils'
 export * from './hooks'
+export * from './database'
```
Comment (Jack-Works) [inline packages/shared/src/index.ts:13]: Well database is the inner implementation detail of the Mask. It should not be shared.

Comment (nuanyang233) [reply packages/shared/src/index.ts:13]: Currently just moving the type to shared.

**Comment by nuanyang233** [comment]
Hmm, this is strange. I don't understand why type-coverage would fail to detect. How should i fix it? @Jack-Works

**Comment by Jack-Works** [comment]
@nuanyang233 please rebase master, install prettier 2.3 and format the code again

**Review by septs — CHANGES_REQUESTED** [review]
plz clear eslint warning, thanks

**Review by Jack-Works — DISMISSED** [review]

File: packages/dashboard/src/components/DashboardFrame/Navigation.tsx [hunk 9]
```diff
@@ -68,7 +87,7 @@ export function Navigation({}: NavigationProps) {
                 </ListItemIcon>
                 <ListItemText primary={t.personas()} />
             </ListItemLink>
-            <ListItem button selected={!!useRouteMatch(Routes.Wallets)} onClick={toggleNavigationExpand}>
+            <ListItem button onClick={toggleNavigationExpand}>
```
Comment (nuanyang233) [inline packages/dashboard/src/components/DashboardFrame/Navigation.tsx:90]: Why? When the route path is `/wallets` the asset page is displayed and the ListItem needs to be selected

File: packages/dashboard/src/components/DashboardFrame/Navigation.tsx [hunk 10]
```diff
@@ -23,15 +23,34 @@ function ListItemLinkUnStyled({ nested, ...props }: LinkProps & ListItemProps &
...
-    [`&.${listItemClasses.root}`]: {
-        paddingLeft: nested ? theme.spacing(9) : theme.spacing(2),
-    },
-    [`&.${listItemClasses.selected}`]: {
-        backgroundColor: 'transparent',
-        borderRight: '4px solid ' + (theme.palette.mode === 'light' ? theme.palette.action.selected : 'white'),
-    },
-}))
+const ListItemLink = styled(ListItemLinkUnStyled)(({ theme, nested }) => {
+    const highlightColor = theme.palette.mode === 'light' ? theme.palette.primary.main : '#fff'
```
Comment (nuanyang233) [inline packages/dashboard/src/components/DashboardFrame/Navigation.tsx:27]: You can use the `MaskColorVar.linkText` to replace this

File: packages/dashboard/src/components/DashboardFrame/Navigation.tsx [hunk 11]
```diff
@@ -23,15 +23,34 @@ function ListItemLinkUnStyled({ nested, ...props }: LinkProps & ListItemProps &
...
+    return {
+        [`&.${listItemClasses.root}`]: {
+            color: theme.palette.mode === 'light' ? '' : 'rgba(255,255,255,.8)',
+            paddingLeft: nested ? theme.spacing(9) : theme.spacing(2),
+        },
+        [`&.${listItemClasses.selected}`]: {
+            color: highlightColor,
+            backgroundColor: 'transparent',
+            position: 'relative',
+            '.MuiListItemIcon-root': {
```
Comment (nuanyang233) [inline packages/dashboard/src/components/DashboardFrame/Navigation.tsx:37]: please use `listItemIconClasses.root` to replace this,  This current practice is not a strong type

File: packages/dashboard/src/components/DashboardFrame/Navigation.tsx (hunk 9 above)
Comment (Jack-Works) [reply packages/dashboard/src/components/DashboardFrame/Navigation.tsx]: Should we revert this?

Comment (Jack-Works) [reply packages/dashboard/src/components/DashboardFrame/Navigation.tsx]: This PR is now in the review process and should not add new non-review changes cc @yanzhihong23

Comment (nuanyang233) [reply packages/dashboard/src/components/DashboardFrame/Navigation.tsx]: I've revert

**Review by Jack-Works — APPROVED** [review]

**Comment by Jack-Works** [comment]
@septs @guanbinrui please review!

**Pull request merged**